mysticeti-core/src/synchronizer.rs

```diff
@@ -333,17 +371,33 @@ where
         &self,
         except: &[AuthorityIndex],
     ) -> Option<(AuthorityIndex, mpsc::Permit<NetworkMessage>)> {
-        let mut senders = self
+        static MILLIS_IN_MINUTE: u128 = Duration::from_secs(60).as_millis();
+        // Exclude synchronizing from anyone that has a latency of over 2 seconds.
+        static CUT_OFF_MILLIS: u128 = Duration::from_secs(2).as_millis();
+        let senders = self
             .senders
             .iter()
-            .filter(|&(index, _)| !except.contains(index))
+            .filter(|&(index, (_, latency_receiver))| {
+                !except.contains(index) && latency_receiver.borrow().as_millis() <= CUT_OFF_MILLIS
```

Comment:
Should we also implement an anti-dos check on the helper nodes to ensure they do not allocate more than a set amount of resources per peer?

Reply:
That's something that we definitely need to do but I would suggest doing it in later iterations as it's a separate domain on its own.